src/ucs/sys/sys.c

```diff
@@ -767,23 +767,42 @@ ucs_status_t ucs_sys_get_proc_cap(uint32_t *effective)
 #endif
 }
 
-static void ucs_sysv_shmget_error_check_EPERM(int flags, char *buf, size_t max)
+static ucs_status_t ucs_sys_get_memlock_privilege(int *lock_priv)
 {
 #if HAVE_SYS_CAPABILITY_H
-    ucs_status_t status;
     uint32_t ecap;
+    ucs_status_t status;
 
     UCS_STATIC_ASSERT(CAP_IPC_LOCK < 32);
     status = ucs_sys_get_proc_cap(&ecap);
-    if ((status == UCS_OK) && !(ecap & UCS_BIT(CAP_IPC_LOCK))) {
-        /* detected missing CAP_IPC_LOCK */
-        snprintf(buf, max, ", CAP_IPC_LOCK privilege is needed for SHM_HUGETLB");
-        return;
+    if(status != UCS_OK) {
```

Comment:
In other places in UCX, it is ignored. Can simply return "false" (0) in such case and make the func signature simpler (no in-out ptr arg).

Reply:
After refactoring according to this suggestion, we'll have the following pattern:
```
  status = ucs_sys_get_proc_cap(&ecap);
  if ((status == UCS_OK) && (ecap & SOME_CAP)) {
```

in 3 places in our code (your PR + 2 existing places).
Maybe make in an inline func?